app/services/ingest_iati_activities.rb

```diff
@@ -0,0 +1,175 @@
+require "nokogiri"
+
+class IngestIatiActivities
+  attr_accessor :file_io, :delivery_partner
+
+  def initialize(delivery_partner:, file_io:)
+    self.delivery_partner = delivery_partner
+    self.file_io = file_io
+  end
+
+  def call
+    doc = Nokogiri::XML(file_io, nil, "UTF-8")
+
+    legacy_activities = doc.xpath("//iati-activity")
+
+    legacy_activities.each do |legacy_activity|
+      ActiveRecord::Base.transaction do
+        new_activity = Activity.new(level: :project, organisation: delivery_partner)
+        add_identifiers(legacy_activity: legacy_activity, new_activity: new_activity)
+
+        parent_activity = find_or_create_parent_activities(delivery_partner: delivery_partner, new_activity: new_activity)
+        new_activity.activity = parent_activity
+
+        add_participating_organisation(delivery_partner: delivery_partner, new_activity: new_activity, legacy_activity: legacy_activity)
+
+        new_activity.title = legacy_activity.elements[2].children.detect { |child| child.name.eql?("narrative") }.children.text if legacy_activity.elements[2].name.eql?("title")
+        new_activity.description = legacy_activity.elements[3].children.detect { |child| child.name.eql?("narrative") }.children.text if legacy_activity.elements[3].name.eql?("description")
+        new_activity.status = legacy_activity.elements.detect { |element| element.name.eql?("activity-status") }.attributes["code"].value
+        new_activity.sector = legacy_activity.elements.detect { |element| element.name.eql?("sector") }.attributes["code"].value
+        new_activity.flow = legacy_activity.elements.detect { |element| element.name.eql?("default-flow-type") }.attributes["code"].value
+        new_activity.aid_type = legacy_activity.elements.detect { |element| element.name.eql?("default-aid-type") }.attributes["code"].value
+
+        add_dates(legacy_activity: legacy_activity, new_activity: new_activity)
+        add_geography(legacy_activity: legacy_activity, new_activity: new_activity)
+        add_transactions(legacy_activity: legacy_activity, new_activity: new_activity)
+
+        new_activity.ingested = true
+        # Set the status to invoke validations
+        new_activity.wizard_status = :complete
```

Comment:
The initial approach for saving and validating is to only create valid records. All mandatory fields must be required before creation in RODA and where data is missing we add default values that a user can identify through the front end and correct before publishing to IATI. Rather than creating records with `save(:validate_false)` which could work but introduces the chances of records being in unexpected states, leading to errors. If we can create valid records by default, we should.